src/main/java/com/list/todo/controllers/AuthController.java

```diff
@@ -0,0 +1,92 @@
+package com.list.todo.controllers;
+
+import java.net.URI;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import com.list.todo.entity.RoleName;
+import com.list.todo.entity.User;
+import com.list.todo.payload.ApiResponse;
+import com.list.todo.payload.JwtAuthenticationResponse;
+import com.list.todo.payload.LoginRequest;
+import com.list.todo.payload.SignUpRequest;
+import com.list.todo.repositories.UserRepository;
+import com.list.todo.security.JwtTokenProvider;
+
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+
+    private AuthenticationManager authenticationManager;
+    private UserRepository userRepository;
+    private PasswordEncoder passwordEncoder;
+    private JwtTokenProvider tokenProvider;
+    
+    @Autowired
+    public AuthController(AuthenticationManager authenticationManager, UserRepository userRepository, 
+    		PasswordEncoder passwordEncoder, JwtTokenProvider tokenProvider) {
+		this.authenticationManager = authenticationManager;
+		this.userRepository = userRepository;
+		this.passwordEncoder = passwordEncoder;
+		this.tokenProvider = tokenProvider;
+	}
+
+	@PostMapping("/signin")
+    public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) {
```

Comment:
indentation please

Reply:
In IDE we don't have this indentation :)